File: marketplace-service/app/src/main/java/com/axonivy/market/model/DeprecationRequest.java
package com.axonivy.market.model;

import com.axonivy.market.enums.PullRequestAction;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class DeprecationRequest {
  private String successorUrl;
  private Boolean isDeprecated;
  private Boolean isAddReadme;
  private PullRequestAction pullRequestAction;
  private String deprecationRequester;
  private Date deprecationDate;
}

File: marketplace-ui/src/app/shared/models/deprecation-request.ts
import { PullRequestAction } from '../enums/pullrequest-action';

export interface DeprecationRequest {
  hasAlternativeExtension?: boolean;
  alternativeExtension?: string;
  successorUrl?: string;
  isDeprecated: boolean | null;
  isAddReadme?: boolean;
  pullRequestAction?: PullRequestAction;
  deprecationDate?: Date | null;
  deprecationRequester?: string;
}
